### ISSUE TITLE
[java] UnsynchronizedStaticFormatter: false negative with `this` pointer

### ISSUE THREAD

**Issue by mohui1999** [pmd/pmd#4413]
**Affects PMD Version:** 6.55.0

**Rule:**[UnsynchronizedStaticFormatter](https://pmd.github.io/latest/pmd_rules_java_multithreading.html#unsynchronizedstaticformatter) 

**Description:**

The `sdf` in code should be synchronized. Adding `this.` to a field should not change the program semantics. In this case, when using `this.sdf.format()` PMD not reports this, while using `sdf.format()` it correctly reports.

**Code Sample demonstrating the issue:**

```java
//False negative case
import java.text.SimpleDateFormat;
public class Foo {
    private static final SimpleDateFormat sdf = new SimpleDateFormat();
    void bar() {
        this.sdf.format();    // should report here, but NOT. False negative
    }
}

// Correctly reported case
import java.text.SimpleDateFormat;
public class Foo {
    private static final SimpleDateFormat sdf = new SimpleDateFormat();
    void bar() {
        sdf.format();   // report here
    }
}
```

**Expected outcome:**
```
UnsynchronizedStaticFormatter:        Static Formatter objects should be accessed in a synchronized manner
```
PMD should report a violation at line 5, but doesn't. This is a false-negative.

**Running PMD through:** *[CLI]*

**Comment by mohui1999** [comment]
Fixed in PMD 7.0
